Fix: Remove unnessary reference to suppress warning

bc-lee opened the pull request:
Apple Clang version 12 (bundled with Xcode 12) complaints about copying
small objects in range loop (-Wrange-loop-analysis introduced by -Wall).
This warning can be easily avoided by removing the reference from
the const pointer type.

LordAro commented:
To be honest, I feel like removing the reference is the better solution here. Taking a reference of a const pointer is fairly ...pointless

TrueBrain commented:
@bc-lee : would you feel up for this? Instead of suppressing the warning, fixing the cause? Would be really awesome if you can fix it :) It is annoying the hell out of me on every CI run :P

bc-lee committed:
Fix: Remove unnessary reference to suppress warning

Apple Clang version 12 (bundled with Xcode 12) complaints about copying
small objects in range loop (-Wrange-loop-analysis introduced by -Wall).
This warning can be easily avoided by removing the reference from
the const pointer type.

bc-lee commented:
@TrueBrain It turns out, fixing this without side effects is almost trivial, so I did it. PTAL.

TrueBrain commented:
Thank you very much, also for the quick reply/fix :D Awesome!
